web3/main.py

```diff
@@ -108,12 +109,15 @@ class Web3:
     isChecksumAddress = staticmethod(is_checksum_address)
     toChecksumAddress = staticmethod(to_checksum_address)
 
-    def __init__(self, providers=empty, middlewares=None, modules=None, ens=empty):
+    def __init__(self, providers=empty, middlewares=None, modules=None, pm=False, ens=empty):
         self.manager = RequestManager(self, providers, middlewares)
 
         if modules is None:
             modules = get_default_modules()
 
+        if pm is True:
+            modules['pm'] = PM
```

Comment:
Not sure if this is the best way to allow "opt-in" usage of the `pm` module? @pipermerriam suggested using `enable_unaudited_features` but it looks like that does nothing currently.

Reply:
You can have a look at the history of `enable_unaudited_features`, before it was changed to do nothing: https://github.com/ethereum/web3.py/blob/838b51d3e73200f7db961b5b6c1a9fc486d947ad/web3/eth.py#L59

The module path/structure is different, `web3.eth.account`, but maybe you could adapt something similar for `web3.pm`.

Reply:
Sounds good, thanks!

Reply:
Another option that is reasonable is to build it like the `Ssh` module. So it's just a free floating module, and if you want to opt-in, you just attach it to your `w3` like:

```
tests/core/shh-module/conftest.py:10:    Shh.attach(web3, "shh")
```

Reply:
Thanks! I'm leaning towards #2 since it reduces code and follows the `ssh` pattern